test/suites/osconfig/lifecycle.robot

```diff
@@ -0,0 +1,45 @@
+*** Settings ***
+Documentation       Tests for MicroShift service lifecycle
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/microshift-process.resource
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           restart
+
+
+*** Variables ***
+${RESTART_ATTEMPTS}     30
+
+
+*** Test Cases ***
+Restarting MicroShift In The Middle Of Startup Should Succeed
+    [Documentation]    Checks if restarting MicroShift during startup fails.
+    ...    (for example due to not stopped microshift-etcd.scope).
+
+    FOR    ${attempt}    IN RANGE    ${RESTART_ATTEMPTS}
+        ${time_to_sleep}=    Evaluate    random.sample(range(1, 40), 1)[0]    random
```

Comment:
Maybe instead of randomly picking values, we could do something more deterministic like start with a 40 second pause, then 30, then 20, 15, 10, 5, 2, 1?

That would even let us define those test cases separately, so they would all be reported individually.

Reply:
PTAL

From log.html perspective: `Restarting MicroShift In The Middle Of Startup Should Succeed` is parent node with following subnodes:
```
- Restart MicroShift 40 Seconds After Starting
- Restart MicroShift 30 Seconds After Starting
... and so on
```

Reply:
Nice, I like the template